mythril/laser/ethereum/state.py

```diff
@@ -64,20 +64,10 @@ def get_word_at(self, index: int):
 
     def __getitem__(self, item):
         if isinstance(item, slice):
-            try:
-                stop = get_concrete_int(item.stop)
-            except AttributeError:
-                stop = item.stop
-
-            if self.concrete and isinstance(stop, int):
-                calldatasize = get_concrete_int(self.calldatasize)
-                if stop > calldatasize:
-                    stop = calldatasize
-
             try:
                 current_index = item.start if isinstance(item.start, BitVecRef) else BitVecVal(item.start, 256)
                 dataparts = []
-                while simplify(current_index != stop):
+                while simplify(current_index != item.stop):
```

Comment:
If item.stop is symbolic, then does this loop run forever?

Reply:
I believe the behavior here is correct.

Apparently the `==` and `!=` operators have different behavior with regards to exceptions, so it's kind of weird. I'm using the `!=` operator, so we're all good.
```
x = BitVec('x', 256)
y = x + 32
z = BitVec('z', 256)

if not simplify(x + 32 != y): print("x + 32 != y") # prints x + 32 != y
if not simplify(x + 32 != z): print("x + 32 != z") # raises Z3Exception("Symbolic expressions cannot be cast to concrete Boolean values.") 
```
The Z3Exception is caught and and IndexError is raised.

Reply:
What if y = x + Bitvec('p', 256)

Reply:
It seems that that also throws a z3 exception